charms/k8s/lib/charms/k8s/v0/k8sd_api_manager.py

```diff
@@ -145,6 +150,96 @@ class CreateJoinTokenResponse(BaseRequestModel):
     metadata: TokenMetadata
 
 
+class ClusterMember(BaseModel):
+    """Represents a member in the k8sd cluster.
+
+    Attributes:
+        Name (str): Name of the cluster member.
+        Address (str): Address of the cluster member.
+        Role (str): Role of the member in the cluster.
+        Fingerprint (str): Fingerprint for the member.
+        Status (str): Current status of the member.
+    """
+
+    Name: str
+    Address: str
+    Role: str
+    Fingerprint: str
+    Status: str
+
+
+class ClusterComponent(BaseModel):
+    """Represents a component in the k8sd cluster.
+
+    Attributes:
+        name (str): Name of the component.
+        status (str): Current status of the component.
+    """
+
+    name: str
+    status: str
+
+
+class ClusterStatus(BaseModel):
+    """Represents the overall status of the k8sd cluster.
+
+    Attributes:
+        Ready (bool): Indicates if the cluster is ready.
+        Members (List[ClusterMember]): List of members in the cluster.
+        Components (List[ClusterComponent]): List of components in the cluster.
+    """
+
+    Ready: bool
+    Members: List[ClusterMember]
+    Components: List[ClusterComponent]
+
+
+class ClusterMetadata(BaseModel):
+    """Metadata containing status information about the k8sd cluster.
+
+    Attributes:
+        status (ClusterStatus): The status of the k8sd cluster.
+    """
+
+    status: ClusterStatus
+
+
+class GetClusterStatusResponse(BaseRequestModel):
+    """Response model for getting the status of the k8sd cluster.
+
+    Attributes:
+        metadata (Optional[ClusterMetadata]): Metadata containing the cluster status.
+                                              Can be None if the status is not available.
+    """
+
+    metadata: Optional[ClusterMetadata] = None
```

Comment:
Why do you need to wrap this here?

The Go type does not have a `metadata` member:
```
// GetClusterStatusResponse is the response for "GET 1.0/k8sd/cluster".
type GetClusterStatusResponse struct {
	ClusterStatus ClusterStatus `json:"status"`
}
```

Reply:
microclusters puts the response in a `metadata` field, along with others to manage errors etc.